logger/src/error.rs

```diff
@@ -0,0 +1,159 @@
+//! Ergonomic log-and-ignore syntax.
+
+/// Trait used to pipeline the common log-and-ignore pattern by
+/// providing better handling of the process, and a clearer more
+/// expressive syntax.
+pub trait Logged {
+    /// Log the provided value. Useful for logging objects in a chain
+    /// of events, or before the `?` operator.
+    #[must_use]
+    fn logged(self) -> Self;
+
+    /// Ignore the provided error.
+    ///
+    /// # Motivation
+    ///
+    /// Sometimes errors cannot be sensibly handled, they can only be
+    /// reported to the user. If the error isn't handled further this
+    /// allows one to cheaply explain to the user of the debug build
+    /// **why** a particular error cannot be ignored.  More
+    /// importantly when one sees an explicit
+    /// ```rust
+    /// err.logged().ignore("I can't handle this error because of X")
+    /// ```
+    /// the reader of the code doesn't need to be provided
+    /// with a comment as to why the error message was ignored
+    #[inline]
+    fn ignored(self, reason: &str)
+    where
+        Self: Sized,
```

Comment:
you should make `Sized` a supertrait of `Logged`, it's not going to be possible to make it a trait object anyhow

Reply:
I'm not sure we gain much. I wonder if this would allow us to work with references separately from the type, which can be interesting.